docs/openfaas-pro/iam/function-authentication.md

```diff
@@ -0,0 +1,275 @@
+# Function Authentication
+
+With OpenFaaS Identity and Access Management (IAM) you can secure function endpoints without having to write any additional code. Access permissions for functions can be configured using IAM Roles and Policies.
+
+!!! note Technical preview
+    Function authentication is currently under technical preview
+
+## Enable function authentication
+
+OpenFaaS Identity and Access Management needs to be enabled and configured in order to use function authentication, see: [Identity and Access Management (IAM)](/openfaas-pro/iam/overview/#installation)
+
+To enable function authentication your functions need to be using a compatible version of the [OpenFaaS watchdog](/architecture/watchdog/). For the classic-watchdog this is any version `> 0.2.3`. The of-watchdog support function authentication for releases `> 0.9.15`.
+
+Authentication can be enabled on a per function basis by setting the `jwt_auth` environment variable to true in the function configuration, e.g:
+
+```yaml
+functions:
+    figlet:
+        skip_build: true
+        image: ghcr.io/openfaas/figlet:latest
+        environment:
+            jwt_auth: true
+```
+
+### Define Roles and Policies
+
+Create a new policy or update your existing policies and include the `Function:Invoke` action to allow function invocations.
+
+The following policy will allow the `env` function in the `dev` namespace to be invoked, and any function in the `openfaas-fn` namespace:
+
+```yaml
+apiVersion: iam.openfaas.com/v1
+kind: Policy
+metadata:
+  name: invoke-policy
+  namespace: openfaas
+spec:
+  statement:
+  - sid: 1-invoke-policy
+    action:
+      - "Function:Invoke"
+    effect: Allow
+    resource:
+      - "openfaas-fn:*"
+      - "dev:env"
+
+```
+
+Create a Role to map the `invoke-policy` to a user or group:
+
+```yaml
+apiVersion: iam.openfaas.com/v1
+kind: Role
+metadata:
+  name: invoke-role
+  namespace: openfaas
+spec:
+  policy:
+  - invoke-policy
+  condition:
+    principal:
+      jwt:sub:
+        - a9e0e67a-5758-4373-a4ba-23957fa66e6b
+    StringEqual:
+      jwt:iss: ["https://keycloak.example.com/realms/openfaas"]
+```
+
+
+## Invoke authenticated functions
+
+To invoke a function that has authentication enabled you need to obtain a function access token. Function access token can be obtained by exchanging a valid ID token for a function access token through the OpenFaaS Gateways token exchange endpoint.
```

Comment:
Function access token => Function access tokens

or

A function access token

I'd like us to use consistent language, in https://www.openfaas.com/blog/walkthrough-iam-for-openfaas/ we talk about an "OpenFaaS Token" 

Here we say: "OpenFaaS access token" - https://www.openfaas.com/blog/kubernetes-tokens-openfaas-api/

Here, we're more specific "OpenFaaS API Access Token" - https://www.openfaas.com/blog/built-in-function-authentication/

Reply:
I rewrote this section to consistently refer to the different types of tokens with the same terminology.